agent/util-scripts/pbench-tool-meister-start

```diff
@@ -0,0 +1,542 @@
+                    # in containers so this does not make sense going forward.
+                    continue
+                tool = tdirent
+                with (self.tg_dir / host / tool).open("r") as fp:
+                    tool_opts = fp.read()
+                if tool not in self.toolnames:
+                    self.toolnames[tool] = {}
+                self.toolnames[tool][host] = tool_opts
+
+    def get_tools(self, host):
+        """Given a target host, return a dictionary with the list of tool names
+        as keys, and the values being their options for that host.
+        """
+        tools = dict()
+        for tool, opts in self.toolnames.items():
+            try:
+                host_opts = opts[host]
+            except KeyError:
+                # This host does not have this tool registered, ignore.
+                pass
+            else:
+                tools[tool] = host_opts
+        return tools
+
+
+def wait_for_subs(chan, expected_tms, logger):
+    """wait_for_subs - Wait for the data sink and the proper number of TMs to
+    register, and when they are all registered, return a dictionary of the
+    data sink and tool meister(s) pids.
+    """
+    pids = dict()
+    have_ds = False
+    num_tms = 0
+    for payload in chan:
+        try:
+            json_str = payload["data"].decode("utf-8")
+        except Exception:
+            logger.warning("data payload in message not UTF-8, '%r'", json_str)
+            continue
+        logger.debug('channel payload, "%r"', json_str)
+        try:
+            data = json.loads(json_str)
+        except json.JSONDecodeError:
+            logger.warning("data payload in message not JSON, '%s'", json_str)
+            continue
+        # We expect the payload to look like:
+        #   { "kind": "<ds|tm>",
+        #     "hostname": "<hostname>",
+        #     "pid": "<pid>"
+        #   }
+        # Where 'kind' is either 'ds' (data-sink) or 'tm' (tool-meister),
+        # 'hostname' is the host name on which that entity is running, and
+        # 'pid' is that entity's PID on that host.
+        try:
+            new_data = dict(
+                kind=data["kind"], hostname=data["hostname"], pid=data["pid"]
+            )
+        except KeyError:
+            logger.warning("unrecognized data payload in message, '%r'", data)
+            continue
+        else:
+            if new_data["kind"] == "ds":
+                pids["ds"] = new_data
+                have_ds = True
+            elif new_data["kind"] == "tm":
+                if "tm" not in pids:
+                    pids["tm"] = []
+                pids["tm"].append(new_data)
+                num_tms += 1
+            else:
+                logger.warning("unrecognized 'kind', in data payload '%r'", data)
+                continue
+        if have_ds and num_tms == expected_tms:
+            break
+    return pids
+
+
+def kill_redis_server(pid_file):
```

Comment:
Wouldnt it be easier to subprocess to pkill here?

Reply:
The `redis-server` has daemonized at this point.